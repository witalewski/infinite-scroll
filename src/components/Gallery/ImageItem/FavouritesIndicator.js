import React from 'react';
import styled from '@emotion/styled';

const FavouritesIndicatorStyled = styled.div`
  position: absolute;
  bottom: 12px;
  right: 12px;
  transition: opacity 0.5s 0.2s;

  &.favourites-indicator--hidden {
    opacity: 0;
  }
`;
export const FavouritesIndicator = ({ isFavourite }) => (
  <FavouritesIndicatorStyled
    className={`favourites-indicator ${!isFavourite &&
      'favourites-indicator--hidden'}`}
  >
    <span role="img" aria-label="favourite">
      ❤️
    </span>
  </FavouritesIndicatorStyled>
);
