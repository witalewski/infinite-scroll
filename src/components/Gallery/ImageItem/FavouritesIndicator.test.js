import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { FavouritesIndicator } from './FavouritesIndicator';

describe('FavouritesIndicator', () => {
  it('matches snapshot for non-favourite', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FavouritesIndicator isFavourite={false} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot for favourite', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FavouritesIndicator isFavourite={true} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
