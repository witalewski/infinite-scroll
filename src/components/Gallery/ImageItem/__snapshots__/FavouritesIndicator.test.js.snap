// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesIndicator matches snapshot for favourite 1`] = `
<ForwardRef(render)
  className="favourites-indicator false"
>
  <span
    aria-label="favourite"
    role="img"
  >
    ❤️
  </span>
</ForwardRef(render)>
`;

exports[`FavouritesIndicator matches snapshot for non-favourite 1`] = `
<ForwardRef(render)
  className="favourites-indicator favourites-indicator--hidden"
>
  <span
    aria-label="favourite"
    role="img"
  >
    ❤️
  </span>
</ForwardRef(render)>
`;
